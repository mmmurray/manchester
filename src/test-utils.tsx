import * as RTL from '@testing-library/react'
import React, { FC } from 'react'
import { ThemeProvider } from 'styled-components'
import { defaultTheme } from './theme'

const Wrapper: FC = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children as any}</ThemeProvider>
)

const render = (el: React.ReactElement) => RTL.render(<Wrapper>{el}</Wrapper>)

export * from '@testing-library/react'
export { render, Wrapper }
