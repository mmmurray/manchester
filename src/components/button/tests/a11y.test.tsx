import React from 'react'
import Button from '..'
import { Wrapper } from '../../../test-utils'

test('default button a11y', async () => {
  await expect(
    <Wrapper>
      <Button>test button</Button>
    </Wrapper>,
  ).toMatchA11ySnapshot()
})
