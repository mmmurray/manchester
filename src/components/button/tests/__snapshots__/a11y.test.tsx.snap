// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default button a11y: toMatchA11ySnapshot 1`] = `
"role: WebArea
children:
  -
    role: button
    name: 'test button'
"
`;

exports[`full width button a11y: toMatchA11ySnapshot 1`] = `
"role: WebArea
children:
  -
    role: button
    name: 'test button'
"
`;
